### PR TITLE
Feature 1

### PR DESCRIPTION
**Pull request by MiguelArgentina**
## In this project we implement authorization to our app
- Project specifications can be found in [The Odin Project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)
- This project implements authentication systems so users can only access areas and content of a site they are authorized to
- AUthentication was implemented using [Devise gem](https://github.com/heartcombo/devise)
- Tests were run to check the app before creating this PR
- We used application layout for sign-in and sign-out links
- Users can create their account using the `Become a member` link
- Users can have many posts but posts only belong to one user

### CODE REVIEW

File: app/views/layouts/_header.html.erb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                <li class="divider"></li>
+                <li>
+                  <%= link_to 'Logout',  destroy_member_session_path, method: :delete %>
+                </li>
+              </ul>
+            </li>
+          <% else %>
+            <li><%= link_to 'Log In', new_member_session_path %></li>
+            <li><%= link_to 'Become a Member', new_member_registration_path %></li>
+          <% end %>
```
Comment (lekegitrepo) [inline app/views/layouts/_header.html.erb]: - remove logic from the view

File: app/views/posts/_errors.html.erb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    <div class="alert alert-danger w-50">
+      The form contains <%= pluralize(@post.errors.count, "error") %>
+    </div>
+    <ul>
+      <% @post.errors.full_messages.each do |error| %>
+        <li class="text-danger"> <%= error %> </li>
+      <% end %>
+    </ul>
+  </div>
+<% end %>
```
Comment (lekegitrepo) [inline app/views/posts/_errors.html.erb:12]: - remove logic from the view

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+          <div class="author panel-body">
+            <%= post.body %>
+            <% if member_signed_in? %>
+              <p>
+                <%= current_member.name %> &nbsp;&bull; <%= post.created_at %>
+              </p>
+            <% end %>
+          </div>
+      </div>
+      <% end %>
```
Comment (lekegitrepo) [inline app/views/posts/index.html.erb]: - remove logic from the view

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required! :recycle:
### Hi @MiguelArgentina and @george-swift :clap: :clap:
#### Well done :muscle: you done a good job, the project looks great.
#### However, some requirements are missing/not implemented correctly. Check my comments below :point_down:

#### 1) Readme:
 - Add instruction on how to install dependencies and also database migration.

#### 2) View & Logic:
 - There should be NO application logic in the views use the helper method instead. Check these links for details [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code), [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers),

#### 3) Routes:
 - The application routes should be defined specifically for the used routes: `resources :posts, only: [:new, :create, :index]`

#### 4) PostController:
 - `new` action method should initialize an empty post `@post = Post.new`

#### _Reviewed by Temitayo Adeleke_

File: app/views/posts/_errors.html.erb (hunk 2 above)
Comment (george-swift) [reply app/views/posts/_errors.html.erb:12]: Requirements #2, #3 and #4 have been implemented. Helpers are particularly for removing complex logic from the views. Views should have as little embedded dynamic code as possible for clean, easy to modify and maintain code. However, the error messages are in a partial that is rendered in line 7 of `posts/new.html.erb`.  Writing another helper method in this case would only generate more markup.


Error_partial

**Review by oloomoses — CHANGES_REQUESTED** [review]
## Changes Required :recycle: 
**Hello team, you have done very well implementing the changes suggested by @lekegitrepo :muscle: 
Your project is amazing, however, there are some critical issues to fix before it can be approved :point_down:**
- [x] As it is, a signed-in user is not able to sign-out. Kindly fix this.
- [x] `.idea` file is not part of the project file and should be moved to the `.gitignore` file

Kindly work on the above suggestions before submitting another review.
Feel free to leave a comment by mentioning me incase of doubts or concerns
_Slack: Oloo Moses_

**Comment by george-swift** [comment]
@oloomoses A signed-in user can click on the `Account` link in the navbar. In the drop-down, there is a link to log out. Both sign-in and sign-out attempts if successful will trigger the devise alert notice. If a signed-in user wishes to cancel their account, they can access the cancel my account option in `Edit profile` still in the dropdown. Again, a successful attempt will trigger a respective alert.


sign-in
sign-out

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# STATUS: Changes Required :red_circle: 

### Hello team, great job implementing all the changes asked previously but still one of them has to be implemented which was asked previously:
- [x] The `Account` dropdown is not working therefore a user cannot logout so please fix this

Please fix the required changes and ask for another code review whenever you're ready

_**If you have any question related to the required changes feel free to text me on slack @GzimAsani** or just add a comment here_

HAPPY CODING :computer: :desktop_computer:

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**